tests/test_hydrology.py

```diff
@@ -23,6 +28,37 @@ def test_simple(self, q_series):
         np.testing.assert_array_equal(out, 2)
 
 
+class TestStandardizedStreamflow:
+    @pytest.mark.slow
+    def test_3d_data_with_nans(self, open_dataset):
+        nc_ds = Path("Raven", "q_sim.nc")
+        # test with data
+        ds = open_dataset(nc_ds)
+        q = ds.q_obs.sel(time=slice("2008")).rename("q")
+        qMM = convert_units_to(q, "mm**3/s", context="hydro")  # noqa
+        # put a nan somewhere
+        qMM.values[10] = np.nan
+        q.values[10] = np.nan
+
+        out1 = land.standardized_streamflow_index(
+            q,
+            freq="MS",
+            window=1,
+            dist="genextreme",
+            method="APP",
+            fitkwargs={"floc": 0},
+        )
+        out2 = land.standardized_streamflow_index(
+            qMM,
+            freq="MS",
+            window=1,
+            dist="genextreme",
+            method="APP",
+            fitkwargs={"floc": 0},
+        )
+        np.testing.assert_array_almost_equal(out1, out2, 3)
```

Comment:
Not sure I understand what is checked here, beyond unit changes.

Reply:
I didn't write the test, Adrien did. But looking at xclim code, this is how many `test_3d_data_with_nans` start. In this case, there is no nan at the end to check specifically if a nan is kept in place, because with the monthly resampling there is no more nan, so it becomes a bit dull indeed. I will add a daily computation with a nan insertion, and check that the SSI is a nan as expected.

The values of the output are already tested in `test_indices`